source/NVDAObjects/IAccessible/ia2TextMozilla.py

```diff
@@ -307,7 +307,7 @@ def _getText(self, withFields, formatConfig=None):
 		while fields[-1] is not None:
 			# The end hasn't yet been reached, which means it isn't a descendant of obj.
 			# Therefore, continue from where obj was embedded.
-			if withFields:
+			if withFields and controlStack:
 				field = controlStack.pop()

```

Comment:
```suggestion
				try:
					field = controlStack.pop()
				except IndexError:
					# We're trying to walk up past our root. This can happen if a descendant
					# object within the range died, in which case _iterRecursiveText will
					# never reach our end object and thus won't yield None. This means this
					# range is invalid, so just return nothing.
					log.debugWarning("Tried to walk up past the root. Objects probably dead.")
					return []
```

Reply:
Is there any way we can confirm that the object is dead in this case?

Reply:
Maybe, but it's complicated and it'd involve tweaking quite a few layers, which has a high regression risk. We'd have to not catch exceptions (i.e. allow them to be raised) in IA2TextTextInfo._getFormatFieldAndOffsets, OffsetsTextInfo.getTextWithFields, IA2TextTextInfo._getTextRange and maybe others.  We'd then need to modify anything which calls those things to expect and gracefully handle the exceptions.

One other thought I had was to check if `_iterTextWithEmbeddedObjects` returns an empty list, and if it does, assume the object is dead. However, I worry there's the real possibility that a descendant object might actually not have any content for some reason; it does happen.